lib/ukstore/Makefile.uk

```diff
@@ -1,53 +1,19 @@
 $(eval $(call addlib_s,libukstore,$(CONFIG_LIBUKSTORE)))
 
-$(call mk_sub_libbuild_dir,libukstore,include/uk/bits)
-
-LIBUKSTORE_GEN_INCLUDES_PATH := $(call sub_libbuild_dir,libukstore,include)
-
 CINCLUDES-y                    += -I$(LIBUKSTORE_BASE)/include
-CINCLUDES-$(CONFIG_LIBUKSTORE) += -I$(LIBUKSTORE_GEN_INCLUDES_PATH)
+CINCLUDES-$(CONFIG_LIBUKSTORE) += -I$(LIBUKSTORE_BUILD)/include
 CXXINCLUDES-y                    += -I$(LIBUKSTORE_BASE)/include
-CXXINCLUDES-$(CONFIG_LIBUKSTORE) += -I$(LIBUKSTORE_GEN_INCLUDES_PATH)
-
-LIBUKSTORE_PHONY_SRC     := $(LIBUKSTORE_BUILD)/uk_libs.in
-LIBUKSTORE_PHONY_SRC_NEW := $(addsuffix .new, $(LIBUKSTORE_PHONY_SRC))
-
-LIBUKSTORE_GEN_SRC := $(LIBUKSTORE_GEN_INCLUDES_PATH)/uk/bits/store_libs.h
-LIBUKSTORE_GEN_ARR := $(LIBUKSTORE_GEN_INCLUDES_PATH)/uk/bits/store_array.h
-LIBUKSTORE_GEN_LD  := $(LIBUKSTORE_BUILD)/store_libs.lds.S
-
-UK_PREPROCESS-$(CONFIG_LIBUKSTORE) += $(LIBUKSTORE_PHONY_SRC) $(LIBUKSTORE_GEN_SRC) $(LIBUKSTORE_GEN_LD) $(LIBUKSTORE_GEN_ARR)
-
-$(LIBUKSTORE_PHONY_SRC): %: %.new
-	@cmp -s $^ $@; if [ $$? -ne 0 ]; then cp $^ $@; fi
-
-$(LIBUKSTORE_BUILD)/uk_libs.in.new:
-	$(call build_cmd,GEN,libukstore,$(notdir $@), \
-		echo $(foreach P,$(UK_PLATS) $(UK_PLATS-y),\
-		     $(if $(call qstrip,$($(call uc,$(P))_LIBS) $($(call uc,$(P))_LIBS-y)),\
-		     $(foreach L,$($(call uc,$(P))_LIBS) $($(call uc,$(P))_LIBS-y), \
-		     $(if $(call qstrip,$($(call vprefix_lib,$(L),SRCS)) $($(call vprefix_lib,$(L),SRCS-y))), \
-		     $(L) \
-		     )))) \
-		     $(UK_LIBS) $(UK_LIBS-y) \
-		| tr ' ' '\n' > $@)
-
-$(LIBUKSTORE_GEN_INCLUDES_PATH)/uk/bits/store_libs.h: $(LIBUKSTORE_BASE)/gen_libs.h.awk $(LIBUKSTORE_BUILD)/uk_libs.in
-	$(call build_cmd,AWK,libukstore,$(notdir $@), \
-		$(AWK) -F '-' -f $(LIBUKSTORE_BASE)/gen_libs.h.awk \
-		$(LIBUKSTORE_BUILD)/uk_libs.in > $@)
-
-$(LIBUKSTORE_BUILD)/store_libs.lds.S: $(LIBUKSTORE_BASE)/gen_libs.ld.awk $(LIBUKSTORE_BUILD)/uk_libs.in
-	$(call build_cmd,AWK,libukstore,$(notdir $@), \
-		$(AWK) -F '-' -f $(LIBUKSTORE_BASE)/gen_libs.ld.awk \
-		$(LIBUKSTORE_BUILD)/uk_libs.in > $@)
-
-$(LIBUKSTORE_GEN_INCLUDES_PATH)/uk/bits/store_array.h: $(LIBUKSTORE_BASE)/gen_array.h.awk $(LIBUKSTORE_BUILD)/uk_libs.in
-	$(call build_cmd,AWK,libukstore,$(notdir $@), \
-		$(AWK) -F '-' -f $(LIBUKSTORE_BASE)/gen_array.h.awk \
-		$(LIBUKSTORE_BUILD)/uk_libs.in > $@)
-
+CXXINCLUDES-$(CONFIG_LIBUKSTORE) += -I$(LIBUKSTORE_BUILD)/include
+
+# Cross-reference libraries catalog generated by lib/uklibid
+LIBUKSTORE_LIBRARIES_IN = $(call sub_build_dir,libuklibid)/libraries.in
```

Comment:
Overall this change is good and cleans up a lot of clutter, one thing that worries me slightly is whether `libraries.in` is a part of `uklibid`'s inner workings, ie considered private. On the other hand I can't think of a better alternative to solve this problem right now.

Reply:
I agree with you one hundred percent. It is probably a good idea to just document this as bad practice but still best in this case. I will add such a comment.